src/main/kotlin/org/treeWare/proto3/message/EncodeProto3.kt

```diff
@@ -0,0 +1,214 @@
+            val resolvedEntity: EntityModel =
+                if (fieldType == FieldType.COMPOSITION) {
+                    checkNotNull(getMetaModelResolved(leaderFieldMeta1)?.compositionMeta)
+                } else {
+                    checkNotNull(getMetaModelResolved(leaderFieldMeta1)?.enumerationMeta)
+                }
+
+            val packageName = getPackageName(resolvedEntity)
+            if (packageName != currentPackage) {
+                val importLine = "import \"$packageName.proto\";"
+                if (!headerWriter.toString().contains(importLine))
+                    headerWriter.writeln("import \"$packageName.proto\";")
+            }
+        }
+        return TraversalAction.CONTINUE
+    }
+
+    override fun leaveFieldMeta(leaderFieldMeta1: EntityModel) {
+    }
+
+    fun visitMessageOptions(leaderFieldMeta: EntityModel) {
+        getProto3MetaModelMap(leaderFieldMeta)?.options?.forEach { option ->
+            bodyWriter.writeln("option $option;")
+        }
+    }
+
+    fun visitFieldOptions(leaderFieldMeta: EntityModel) {
+        var hasWrittenFirstOption = false
+
+        if (isKeyFieldMeta(leaderFieldMeta)) {
+
+            keyOptions.forEach { option ->
+                val prefix = when (hasWrittenFirstOption) {
+                    false -> {
+                        hasWrittenFirstOption = true
+                        " ["
+                    }
+                    true -> ", "
+                }
+                bodyWriter.write("$prefix$option")
+            }
+        } else if (isRequiredFieldMeta(leaderFieldMeta)) {
+            requiredOptions.forEach { option ->
+                val prefix = when (hasWrittenFirstOption) {
+                    false -> {
+                        hasWrittenFirstOption = true
+                        " ["
+                    }
+                    true -> ", "
+                }
+                bodyWriter.write("$prefix$option")
+            }
+        }
+
+        getProto3MetaModelMap(leaderFieldMeta)?.options?.forEach { option ->
+            val prefix = when (hasWrittenFirstOption) {
+                false -> {
+                    hasWrittenFirstOption = true
+                    " ["
+                }
+                true -> ", "
+            }
+            bodyWriter.write("$prefix$option")
+        }
+        if (hasWrittenFirstOption) bodyWriter.write("]")
+        bodyWriter.writeln(";")
+    }
+
+    fun visitImports(leaderMeta: MainModel) {
+        getProto3MetaModelMap(leaderMeta)?.imports?.forEach { import ->
+            val importString = "import \"$import\""
+            imports.add(importString)
+        }
+    }
+
+    fun writeImports() {
+        imports.forEach {
+            bodyWriter.writeln(it)
+        }
+        bodyWriter.writeln("")
+    }
+
+    fun visitKeyFieldOptions(leaderMeta: MainModel) {
+        getProto3MetaModelMap(leaderMeta)?.keyFieldOptions?.forEach { keyOption ->
```

Comment:
nit: body on the same line, and use `it` instead of named `keyOption` lambda parameter.